pkg/leaderelection/leaderelection.go

```diff
@@ -0,0 +1,53 @@
+package leaderelection
+
+import (
+	"time"
+
+	"github.com/golang/glog"
+	configv1 "github.com/openshift/api/config/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/utils"
+)
+
+const (
+	// Defaults follow conventions
+	// https://github.com/openshift/enhancements/blob/master/CONVENTIONS.md#high-availability
+	// Impl Calculations: https://github.com/openshift/library-go/commit/7e7d216ed91c3119800219c9194e5e57113d059a
+	defaultLeaseDuration = 137 * time.Second
+	defaultRenewDeadline = 107 * time.Second
+	defaultRetryPeriod   = 26 * time.Second
+)
+
+func GetLeaderElectionConfig(c client.Client, enabled bool) (defaultConfig configv1.LeaderElection) {
+	defaultConfig = configv1.LeaderElection{
+		Disable:       !enabled,
+		LeaseDuration: metav1.Duration{Duration: defaultLeaseDuration},
+		RenewDeadline: metav1.Duration{Duration: defaultRenewDeadline},
+		RetryPeriod:   metav1.Duration{Duration: defaultRetryPeriod},
+	}
+
+	if enabled {
+		isSno, err := utils.IsSingleNodeCluster(c)
```

Comment:
isSingleNode ?

if sno stands for single node openshift, thats not the case here as this bit will run on vanilla k8s as well

Reply:
Updated as well